tsdb/querier.go

```diff
@@ -1098,38 +1268,56 @@ func (it *chunkSeriesIterator) resetCurIterator() {
 	it.cur = it.bufDelIter
 }
 
-func (it *chunkSeriesIterator) Seek(t int64) (ok bool) {
+func (it *chunkSeriesIterator) Seek(t int64) bool {
+	if it.Err() != nil || it.i > len(it.chunks)-1 {
+		return false
+	}
+
 	if t > it.maxt {
+		// Exhaust iterator.
+		it.i = len(it.chunks)
```

Comment:
maybe it is just my style, but I would think it will be easier to follow with an explicit var `it.exhausted = true`. Can leave as it if you prefer.